src/pineko/scale_variations.py

```diff
@@ -0,0 +1,266 @@
+"""Module to generate scale variations."""
+import pathlib
+from typing import Tuple
+
+import numpy as np
+import pineappl
+import rich
+from eko import beta
+
+from . import check
+
+AS_NORM = 1.0 / (4.0 * np.pi)
+OrderTuple = Tuple[int, int, int, int]
+"""Tuple representing a PineAPPL order."""
+
+
+def sort_qcd_orders(order: OrderTuple) -> int:
+    """Define the sorting for an order. In particular the order is sorted according to the as order."""
+    return order[0]
+
+
+def ren_sv_coeffs(m, max_as, logpart, which_part, nf):
+    """Return the ren_sv contribution relative to the requested log power and perturbative order contribution (which_part).
+
+    Parameters
+    ----------
+    m : int
+        first non zero perturbative order
+    max_as : int
+        max order of alpha_s
+    logpart : int
+        power of the renormalization scale log asked
+    which_part : int
+        asked perturbative order contribution to be rescaled
+    nf : int
+        number of active flavors
+
+    Returns
+    -------
+    float
+        renormalization scale variation contribution
+    """
+    bcoeff = beta.beta_qcd((max_as - logpart - which_part + 2, 0), nf)
+    as_norm_fact = AS_NORM ** (max_as - which_part)
+    m_factor = m
+    if max_as == 0:
+        return 0.0
+    elif max_as == 2:
+        if which_part > 0:
+            m_factor += 1
+        else:
+            if logpart > 1:
+                m_factor = 0.5 * m_factor * (m_factor + 1)
+    return m_factor * as_norm_fact * bcoeff
+
+
+def compute_orders_map(m, max_as, min_al):
+    """Compute a dictionary with all the necessary orders to compute to have the full renormalization scale variation.
+
+    Parameters
+    ----------
+    m : int
+        first non zero perturbative order of the grid
+    max_as : int
+        max alpha_s order
+    min_al : int
+        al order of leading order
+    Returns
+    -------
+    dict(tuple(int))
+        description of all the needed orders
+    """
+    orders = {}
+    for delt in range(max_as):
+        orders[(m + max_as, min_al, delt + 1, 0)] = [
+            (m + de, min_al, 0, 0) for de in range(max_as - delt)
+        ]
+    return orders
```

Comment:
In principle even this can be a clean dictionary comprehension:
```suggestion
    return {
        (m + max_as, min_al, delt + 1, 0): [
            (m + de, min_al, 0, 0) for de in range(max_as - delt)
        ]
        for delt in range(max_as)
    }
```

(Most likely Black would reformat it differently...)

Reply:
[4ba538a](https://github.com/NNPDF/pineko/pull/75/commits/4ba538a43a1ed708e52fa1d7294f129107c81f10)